src/main/java/games/strategy/engine/data/TechnologyFrontier.java

```diff
@@ -38,7 +38,7 @@ public void setGameData(final GameData gameData) {
   private void reorderTechsToMatchGameTechsOrder() {
     final GameData gameData = getData();
     if (gameData != null) {
-      Util.reorder(m_techs, gameData.getTechnologyFrontier().getTechs());
+      m_techs.sort(Comparator.comparing(gameData.getTechnologyFrontier().getTechs()::indexOf));
```

Comment:
If you want to continue caching the index, you can do so with a three-liner:
```
final List<T> order = gameData.getTechnologyFrontier().getTechs();
final Map<T, Integer> cache = order.stream().collect(Collectors.toMap(Function.identity(), order::indexOf));
m_techs.sort(Comparator.comparing(it -> cache.getOrElse(it, 0));
```
Or similar...

Reply:
Alternatively you could use one of the various methods in the guava maps class to cache the index values in a map...

Reply:
Given that we're dealing with less than 20 elements here, I still think using a cache is probably overkill.  But when in doubt, measure it. :smile:

I built the following test:

1. Warm up the JVM.
1. Sort a 20-element list using the original algorithm (reorder1) 1,000,000 times.
1. Sort a 20-element list using the algorithm proposed in this PR (reorder2) 1,000,000 times.
1. Sort a 20-element list using the algorithm you proposed above (reorder3) 1,000,000 times.

I ran that test 10 times.  The average run times on my machine are listed below:

Algorithm | Average Test Run Time (ms)
:-- | :--
reorder1 | 2,735
reorder2 | 2,324
reorder3 | 2,626

That's about an 18% spread, which makes them effectively the same in Big-O notation.  So, given that it appears the algorithms have the same performance for the expected conditions under which they'll operate, I'd say let's stick with the most straightforward code. :smile:

Reply:
Thanks for the benchmark. 
I'm convinced now ^^